staging/src/k8s.io/client-go/tools/leaderelection/leaderelection_test.go

```diff
@@ -1212,6 +1231,25 @@ func testReleaseOnCancellation(t *testing.T, objectType string) {
 				t.Fatal("the lock was not released")
 			}
 			wg.Wait()
+			assertEqualEvents(t, test.expectedEvents, recorder.Events)
 		})
 	}
 }
+
+func assertEqualEvents(t *testing.T, expected []string, actual <-chan string) {
+	for _, e := range expected {
+		a := <-actual
```

Comment:
do you think we should try to avoid this hangs if no events exist?

Reply:
IMO it's fine for a test to hang if it's broken. It's easy to debug that way - run under a debugger and see where it's stuck.

Reply:
It's a bit of waste - we'r usually adding timeouts in such places, sth like:
```
for _, e := range expected {
  select {
  case a := <-actual:
     ...
  case <-time.After(wait.ForeverTestTimeout):
      ...
  }
}
```

It doesn't add much complexity and it fails in easier way to debug.

Reply:
Done.